tools/lsp/common/rename_component.rs

```diff
@@ -644,15 +731,87 @@ impl DeclarationNodeQuery {
                 query: self,
             }),
             _ => {
-                let document =
-                    document_cache.get_document_by_path(self.token.source_file.path())?;
+                fn find_declared_identifier_in_element(
+                    query: &DeclarationNodeQuery,
+                    element: &syntax_nodes::Element,
+                ) -> Option<syntax_nodes::DeclaredIdentifier> {
+                    for prop in element.PropertyDeclaration() {
+                        let identifier = prop.DeclaredIdentifier();
+                        if i_slint_compiler::parser::identifier_text(&identifier).as_ref()
+                            == Some(&query.token_info.name)
+                        {
+                            return Some(identifier);
+                        }
+                    }
+                    None
+                }
+
+                if let Some(declared_identifier) = match &self.token_info.info {
+                    common::token_info::TokenInfo::NamedReference(nr) => {
+                        if i_slint_compiler::parser::normalize_identifier(nr.name())
+                            == self.token_info.name
+                        {
+                            nr.element()
+                                .borrow()
+                                .debug
+                                .iter()
+                                .filter_map(|di| {
+                                    find_declared_identifier_in_element(&self, &di.node)
+                                })
+                                .next()
+                        } else {
+                            None
+                        }
+                    }
+                    common::token_info::TokenInfo::IncompleteNamedReference(element_type, name) => {
+                        if name == &self.token_info.name {
+                            match &element_type {
+                                i_slint_compiler::langtype::ElementType::Component(component) => {
+                                    find_declared_identifier_in_element(
+                                        &self,
+                                        &syntax_nodes::Component::from(
+                                            component.node.as_ref()?.clone(),
+                                        )
+                                        .Element(),
+                                    )
+                                }
+                                _ => None,
+                            }
+                        } else {
+                            None
+                        }
+                    }
+                    _ => None,
+                } {
```

Comment:
Style: I find this `if let ... = match { ... } { ... }` a bit confusing, especially when the match span over many lines.
Maybe it would read better with 
```rs
let declared_identifier = match { ... };
if let Some(declared_identitier) = declared_identifier { ... };
```

Also 
```rs
match foo { 
   Bar => { if zoo { ... } else { None }  } 
   ...  
   _ => None
}
```

Can be simplified in 
```rs
match foo {
   Bar if zoo => { ... }
   ...
   _ => None
}
```

And you save one level.
Same with match that only have one branch, they can sometimes look better with an `if let`

Anyway, this is just style. Feel free to ignore this advice.